File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/lock/realm/SaveLockUseCase.java
package com.lattis.ellipse.domain.interactor.lock.realm;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.model.Lock;
import com.lattis.ellipse.domain.repository.LockRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 4/28/17.
 */

public class SaveLockUseCase extends UseCase<Lock>

{

    private LockRepository lockRepository;
    private Lock lock;

    @Inject
    public SaveLockUseCase(ThreadExecutor threadExecutor,
                           PostExecutionThread postExecutionThread,
                           LockRepository lockRepository) {
        super(threadExecutor, postExecutionThread);
        this.lockRepository = lockRepository;
    }

    public SaveLockUseCase withLock(Lock lock) {
        this.lock = lock;
        return this;
    }


    @Override
    protected Observable<Lock> buildUseCaseObservable() {
        return this.lockRepository.createOrUpdateLock(lock);
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/lock/realm/SaveLockUseCase.kt
package com.lattis.domain.usecase.lock.realm

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.LockRepository
import com.lattis.domain.usecase.base.UseCase
import com.lattis.domain.models.Lock
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class SaveLockUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val lockRepository: LockRepository
) : UseCase<Lock>(threadExecutor, postExecutionThread) {
    private lateinit var lock: Lock
    fun withLock(lock: Lock): SaveLockUseCase {
        this.lock = lock
        return this
    }

    override fun buildUseCaseObservable(): Observable<Lock> {
        return lockRepository.createOrUpdateLock(lock)
    }

}